gammapy/cube/basic_cube.py

```diff
@@ -0,0 +1,74 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions to perform basic functions for map and cube analysis.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from ..maps import WcsNDMap
+
+
+def fill_map_counts(event_list, ndmap):
+    """Fill a ``WcsMap` with events from an EventList.
+
+     The energy of the events is used for a non-spatial axis homogeneous to energy.
+     The other non-spatial axis names should have an entry in the colum names of the ``EventList``
+
+     Parameters
+     ----------
+     event_list : `~gammapy.data.EventList`
+             the input event list
+     ndmap : `~gammapy.maps.Map`
+         Target map
+     """
+    # The list will contain the event table entries to be fed into the WcsNDMap
+    coord_dict = dict()
+    ref_geom = ndmap.geom
+
+    # Add sky coordinates to dictionary
+    coord_dict.update(skycoord=event_list.radec)
+
+    # Now check the other axes and find corresponding entries in the EventList
+    # energy and time are specific types
+    # TODO: add proper extraction for time
+    for i, axis in enumerate(ref_geom.axes):
+        if axis.type == 'energy':
+            # This axis is the energy. We treat it differently because axis.name could be e.g. 'energy_reco'
+            coord_dict.update({axis.name: event_list.energy.to(axis.unit)})
+        # We look for other axes name in the table column names (case insensitive)
+        else:
+            try:
+                # Here we implicitly assume that there is only one column with the same name
+                column_name = next(_ for _ in event_list.table.colnames if _.upper() == axis.name.upper())
+                coord_dict.update({axis.name: event_list.table[column_name].to(axis.unit)})
+            except StopIteration:
+                raise ValueError("Cannot find MapGeom axis {} in EventList", axis.name)
+    # Fill it
+    ndmap.fill_by_coord(coord_dict)
+
+
+def make_map_counts(event_list, ref_geom, meta=None):
```

Comment:
I'd suggest to just put "geom", it's clear that "make_map_counts" with a "geom" passed in should make a map with that geometry, no?
Let's decide here about "map" and "geom" argument names in this PR, and then try to be consistent everywhere in Gammapy as much as possible.